### PR TITLE
Add confirm page in rider-app

### PR DESCRIPTION
**Pull request by andyindahouse**
We can use this step in the rider flow to set the discounts too  🤑

Captura de pantalla 2021-07-25 a las 12 53 43

### CODE REVIEW

File: BACKEND-LOCAL-SETUP.md [hunk 1]
```diff
@@ -11,3 +11,5 @@
    [Node: Nodemon](./.vscode/launch.json) one. There you'll be prompted to choose the proper process to attach
    to. Filter them by by searching for "`node inspect`", and select it.
 1. Now you can use breakpoints right in the sourcecode.
+
+## You can access to mongo express interface in http://0.0.0.0:8081/
```
Comment (SergioMorchon) [inline BACKEND-LOCAL-SETUP.md:15]: Maybe just bold? but not a title xD


## You can access to mongo express interface in http://0.0.0.0:8081/

vs.

**You can access to mongo express interface in http://0.0.0.0:8081/**

```markdown
**You can access to mongo express interface in http://0.0.0.0:8081/**
```

File: backend/src/controllers/orders-rider.ts [hunk 2]
```diff
@@ -135,6 +135,44 @@ export default {
             next(err);
         }
     },
+    updateTotalShoppingCartCost: async ({params, body, user}, res, next) => {
+        try {
+            const id = new mongodb.ObjectID(params.id);
+            const deliveryPrice = getDeliveryPrice();
+
+            if (body.totalShoppingCart <= 0) {
+                next(new Error400(`totalShoppinCart cannot be minor that zero`));
```
Comment (SergioMorchon) [inline backend/src/controllers/orders-rider.ts:144]: missing return?

Or given that this is within a try-catch, you can just `throw new Error400...`

File: packages/models/index.d.ts [hunk 3]
```diff
@@ -89,7 +89,14 @@ export interface ShoppingCart {
     readonly totalShoppingCart: number;
 }
 
-export type OrderStatus = 'pending' | 'cancelled' | 'in-progress' | 'comming' | 'finalized' | 'issue';
+export type OrderStatus =
+    | 'pending'
+    | 'cancelled'
+    | 'in-progress'
+    | 'confirm-cost'
+    | 'comming'
```
Comment (SergioMorchon) [inline packages/models/index.d.ts:97]: I just noticed this: it looks like a typo. Should be `coming`, or `in-delivery`

File: rider-app/src/components/product-detail.tsx [hunk 4]
```diff
@@ -185,89 +182,29 @@ const ProductDetail = ({product, closeModal, updateProduct, replaceProduct}: Pro
...
-                            </Typography>
-                        </div>
-                    )}
+                    <div className={classes.infoSection}>
+                        {replaceProduct && (
+                            <div className={classes.infoItem}>
+                                <IonIcon className={classes.icon} size="large" icon={syncOutline} />
+                                <Typography variant="body2">
+                                    Puedes reemplazar este producto por uno equivalente, solo si te has
+                                    asegurado que no está disponible
```
Comment (SergioMorchon) [inline rider-app/src/components/product-detail.tsx:191]: (...) asegurado _de_ que (...) (https://fundeu.do/asegurarse-de-que-mejor-que-asegurarse-que/)

File: rider-app/src/pages/confirm-cost.tsx [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+
+const useStyles = createUseStyles(({palette}) => ({
+    container: {
+        paddingTop: 16,
+    },
+    footer: {
+        padding: 16,
+    },
+    inputBorder: {
+        border: `1px solid ${palette.grey}`,
```
Comment (SergioMorchon) [inline rider-app/src/pages/confirm-cost.tsx:30]: Maybe `palette.divider`?

File: rider-app/src/pages/confirm-cost.tsx [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+    const history = useHistory();
+    const {id = null} = useParams<{id: string}>();
+    const [order, setOrder] = React.useState<OrderModel | null>(null);
+    const [showAlert, setShowAlert] = React.useState(false);
+    const [totalShoppingCart, setTotalShoppingCart] = React.useState<string | null | undefined>(null);
+    const [showError, setShowError] = React.useState(false);
+    const handleConfirmCost = React.useCallback(async () => {
+        if (id && order?._id) {
+            await kiwiApi.updateTotalShoppingCartCost(order._id, Number(totalShoppingCart));
+            await kiwiApi.updateStatusOrder(order?._id || '', 'comming');
```
Comment (SergioMorchon) [inline rider-app/src/pages/confirm-cost.tsx:47]: Better
```typescript
await Promise.all([
  kiwiApi.updateTotalShoppingCartCost(order._id, Number(totalShoppingCart)),
  kiwiApi.updateStatusOrder(order?._id || '', 'comming'),
]);
```
So we await for both in parallel, and not in series

File: rider-app/src/pages/confirm-cost.tsx [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+                                }
+                                setTotalShoppingCart(e.detail.value);
+                            }}
+                            type="number"
+                        />
+                    </div>
+                    {showError && (
+                        <Box padding={8}>
+                            <Typography color={palette.error.main} variant="caption2">
+                                Este campo no puede estar vacio ni ser menor o igual a cero
```
Comment (SergioMorchon) [inline rider-app/src/pages/confirm-cost.tsx:96]: _vacio_ > _vacío_

Also I'd rephrase it to be positive: "Este campo es obligatorio y debe ser mayor a cero"

File: rider-app/src/pages/confirm-cost.tsx [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+                            </Typography>
+                        </Box>
+                    )}
+                </Box>
+
+                <IonAlert
+                    isOpen={showAlert}
+                    onDidDismiss={() => setShowAlert(false)}
+                    cssClass="my-custom-class"
+                    header={'Confirmar el importe de la compra'}
```
Comment (SergioMorchon) [inline rider-app/src/pages/confirm-cost.tsx:106]: Unneeded curly braces

**Review by SergioMorchon — APPROVED** [review]
🚀🌑

File: packages/models/index.d.ts (hunk 3 above)
Comment (andyindahouse) [reply packages/models/index.d.ts:97]: 🤦

File: rider-app/src/pages/confirm-cost.tsx (hunk 6 above)
Comment (andyindahouse) [reply rider-app/src/pages/confirm-cost.tsx:47]: in this case we don't want to change the status if the first one fails

File: rider-app/src/pages/confirm-cost.tsx (hunk 5 above)
Comment (andyindahouse) [reply rider-app/src/pages/confirm-cost.tsx:30]: we need one more dark

File: rider-app/src/pages/confirm-cost.tsx (hunk 6 above)
Comment (SergioMorchon) [reply rider-app/src/pages/confirm-cost.tsx]: Then we should do it in a single call, with a transactional change in the backend

**Review by danicg — APPROVED** [review]
👍

File: rider-app/src/pages/confirm-cost.tsx (hunk 6 above)
Comment (andyindahouse) [reply rider-app/src/pages/confirm-cost.tsx]: jum, you're right, I'll change it
